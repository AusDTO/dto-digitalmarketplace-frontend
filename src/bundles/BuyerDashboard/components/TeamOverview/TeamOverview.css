.teamoverview-main {
    width: 74.48861%
}

.contact-us-exlink {
    border-bottom: 1px solid #5bcbe3;    
}

@media (max-width: 767px) {
    .teamoverview-main {
        width: 100%
    }

    /* Force table to not be like tables anymore */
    table, thead, tbody, td, tr {
        display: block;
    }

    .teamoverview-table-heading {
        display: block;
    }

    /* Hide table headers (but not display: none;, for accessibility) */
    .teamoverview-table-row {
        position: absolute;
        top: -9999px;
        left: -9999px;
    }

    .summary-item-row td {
        /* Behave  like a "row" */
        border: hidden;
        position: relative;
        padding-left: 50%;
    }

    td:before {
        /* Now like a table header */
        position: absolute;
        /* Top/left values mimic padding */
        top: 6px;
        left: 6px;
        width: 45%;
        padding-right: 10px;
        white-space: nowrap;
    }

    .summary-item-body thead {
        border-bottom: 3px solid #717171 !important;
        display: block;
        vertical-align: middle;
    }

    .summary-item-field-name {
        font-weight: bold !important;
        padding-bottom: 0px !important;
        padding-top: 10px !important;
        /*width: 100% !important;*/
    }

    .summary-item-field {
        padding-bottom: 12px !important;
        padding-top: 0px !important;
    }

}