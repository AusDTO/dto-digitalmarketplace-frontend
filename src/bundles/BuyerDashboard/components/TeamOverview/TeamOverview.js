import React from 'react';
import isEmpty from 'lodash/isEmpty';

import './TeamOverview.css'

const TeamOverview = ({members}) => (
  <div styleName="teamoverview-main">
    <h2>Active team members</h2>
    {(isEmpty(members) ?
        "You are the only registered buyer in your team" :
    <table className="content-table summary-item-body">
      <thead>
      <tr styleName="teamoverview-table-row">
        <th styleName="teamoverview-table-heading">Name</th>
        <th styleName="teamoverview-table-heading">Email</th>
      </tr>
      </thead>
      <tbody>
      {members.map((member, i) => {
        return (
          <tr styleName="summary-item-row" key={i}>
            <td styleName="summary-item-field-name">{member.name}</td>
            <td styleName="summary-item-field">
              <a href={"mailto:" + member.email_address}>{member.email_address}</a>
            </td>
          </tr>
        )
      })}
      </tbody>
    </table>
      )}
  </div>
)

TeamOverview.propTypes = {
  members: React.PropTypes.array.isRequired
}

export default TeamOverview