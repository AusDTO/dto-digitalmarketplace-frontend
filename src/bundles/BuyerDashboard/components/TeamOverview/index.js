import TeamOverview from './TeamOverview';

export default TeamOverview;
