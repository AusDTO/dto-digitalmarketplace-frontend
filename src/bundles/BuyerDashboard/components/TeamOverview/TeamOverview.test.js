import React from 'react';
import renderer from 'react-test-renderer';

import sampleState from '../DashboardBriefs/DashboardBriefs.json'
import TeamOverview from './TeamOverview'

test('TeamOverview renders with props', () => {
  const state = Object.assign(sampleState, {})
   const component = renderer.create(
     <TeamOverview members={state.team.members} />
   )
   let tree = component.toJSON()
   expect(tree).toMatchSnapshot()
 })

test('TeamOverview renders with no props', () => {
  const members = []
   const component = renderer.create(
     <TeamOverview members={members} />
   )
   let tree = component.toJSON()
   expect(tree).toMatchSnapshot()
 })
