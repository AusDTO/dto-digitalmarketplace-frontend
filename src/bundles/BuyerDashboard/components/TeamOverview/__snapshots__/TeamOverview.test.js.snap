// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamOverview renders with no props 1`] = `
<div
  styleName="teamoverview-main"
>
  <h1>
    Active team members
  </h1>
  <p>
    If this list contains members who have left your organisation, please 
    <span
      styleName="contact-us-exlink"
    >
      <a
        href="/contact-us"
        rel="external"
        target="_blank"
      >
        contact us
      </a>
    </span>
     to have them removed
  </p>
  You are the only registered buyer in your team
</div>
`;

exports[`TeamOverview renders with props 1`] = `
<div
  styleName="teamoverview-main"
>
  <h1>
    Active team members
  </h1>
  <p>
    If this list contains members who have left your organisation, please 
    <span
      styleName="contact-us-exlink"
    >
      <a
        href="/contact-us"
        rel="external"
        target="_blank"
      >
        contact us
      </a>
    </span>
     to have them removed
  </p>
  <table
    className="content-table summary-item-body"
  >
    <thead>
      <tr
        styleName="teamoverview-table-row"
      >
        <th
          styleName="teamoverview-table-heading"
        >
          Name
        </th>
        <th
          styleName="teamoverview-table-heading"
        >
          Email
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        styleName="summary-item-row"
      >
        <td
          styleName="summary-item-field-name"
        >
          AJ
        </td>
        <td
          styleName="summary-item-field"
        >
          <a
            href="mailto:marketplace@digital.gov.au"
          >
            marketplace@digital.gov.au
          </a>
        </td>
      </tr>
      <tr
        styleName="summary-item-row"
      >
        <td
          styleName="summary-item-field-name"
        >
          Alex
        </td>
        <td
          styleName="summary-item-field"
        >
          <a
            href="mailto:alex.sadleir+s@digital.gov.au"
          >
            alex.sadleir+s@digital.gov.au
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
